src/app/shell/personal-cabinet/provider/not-provider-admin.guard.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, UrlTree } from '@angular/router';
+import { Select } from '@ngxs/store';
+import { Observable } from 'rxjs';
+import { map, filter } from 'rxjs/operators';
+
+import { Role, Subrole } from 'shared/enum/role';
+import { RegistrationState } from 'shared/store/registration.state';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NotProviderAdminGuard implements CanActivate {
+  @Select(RegistrationState.role)
+  public role$: Observable<string>;
+  @Select(RegistrationState.subrole)
+  public subrole$: Observable<string>;
+
+  public canActivate(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return (
+      this.role$.pipe(
```

Comment:
Combine the observable streams into a single observable
you can use `combineLatest` operator or smth similar